rl4co/models/common/constructive/base.py

```diff
@@ -212,6 +212,7 @@ def forward(
             temperature=decoding_kwargs.pop("temperature", self.temperature),
             tanh_clipping=decoding_kwargs.pop("tanh_clipping", self.tanh_clipping),
             mask_logits=decoding_kwargs.pop("mask_logits", self.mask_logits),
+            only_store_selected_logp=not return_entropy,
```

Comment:
This is the only case; right? Otherwise, we could pop this from the kwargs like

```python
kwargs.pop("only_store_selected_logp", not return_entropy)
```

Reply:
nice idea using `pop`, I will adapt the PR accordingly